service/history/ndc/workflow_resetter.go

```diff
@@ -48,14 +48,15 @@ import (
 	"go.temporal.io/server/common/persistence"
 	"go.temporal.io/server/common/primitives/timestamp"
 	"go.temporal.io/server/common/util"
+	"go.temporal.io/server/service/history/api/updateworkflow"
 	"go.temporal.io/server/service/history/consts"
 	"go.temporal.io/server/service/history/shard"
 	"go.temporal.io/server/service/history/workflow"
 	wcache "go.temporal.io/server/service/history/workflow/cache"
 )
 
 type (
-	workflowResetReapplyEventsFn func(ctx context.Context, resetMutableState workflow.MutableState) error
+	workflowResetReapplyEventsFn func(ctx context.Context, resetMutableState workflow.MutableState) ([]*historypb.HistoryEvent, error)
```

Comment:
`ResetWorkflow` now needs to know the events that have been reapplied, so that it can call advance the update state machine for `UpdateRequested` events. Several of the changes in this PR are passing reflect the slice of reapplied events up the stack from the low level `reapplyEvents` function to `ResetWorkflow`.